crates/host-bench/src/lib.rs

```diff
@@ -236,21 +237,46 @@ pub async fn run_reth_benchmark(args: HostArgs, openvm_client_eth_elf: &[u8]) ->
     let leaf_log_blowup = args.benchmark.leaf_log_blowup.unwrap_or(RETH_DEFAULT_LEAF_LOG_BLOWUP);
     args.benchmark.leaf_log_blowup = Some(leaf_log_blowup);
 
-    let vm_config = reth_vm_config(app_log_blowup);
-    let app_config = args.benchmark.app_config(vm_config);
-
     #[cfg(feature = "cuda")]
     println!("CUDA Backend Enabled");
-    let sdk = Sdk::new(app_config.clone())?
+
+    let mut vm_config = reth_vm_config(app_log_blowup);
+    let mut app_config = args.benchmark.app_config(vm_config);
+    let mut sdk = Sdk::new(app_config.clone())?
         .with_agg_config(args.benchmark.agg_config())
         .with_agg_tree_config(args.benchmark.agg_tree_config);
 
+    if args.app_pk_path.is_some() != args.agg_pk_path.is_some() {
+        eyre::bail!("app_pk_path and agg_pk_path must be provided together");
+    }
     if let Some(app_pk_path) = args.app_pk_path {
         let app_pk: AppProvingKey<SdkVmConfig> = read_object_from_file(app_pk_path)?;
-        sdk.set_app_pk(app_pk).map_err(|_| eyre::eyre!("failed to set app pk"))?;
-    }
-    if let Some(agg_pk_path) = args.agg_pk_path {
+        let agg_pk_path = args.agg_pk_path.unwrap();
         let agg_pk: AggProvingKey = read_object_from_file(agg_pk_path)?;
+        vm_config = app_pk.app_vm_pk.vm_config.clone();
+        args.benchmark.app_log_blowup = Some(app_pk.app_fri_params().log_blowup);
+        args.benchmark.leaf_log_blowup = Some(agg_pk.leaf_vm_pk.fri_params.log_blowup);
+        app_config = args.benchmark.app_config(vm_config);
+        let agg_config = AggregationConfig {
+            leaf_fri_params: agg_pk.leaf_vm_pk.fri_params,
+            internal_fri_params: agg_pk.internal_vm_pk.fri_params,
+            root_fri_params: agg_pk.root_verifier_pk.vm_pk.fri_params,
+            profiling: args.benchmark.profiling,
+            compiler_options: CompilerOptions {
+                enable_cycle_tracker: args.benchmark.profiling,
+                ..Default::default()
+            },
+            root_max_constraint_degree: agg_pk
+                .root_verifier_pk
+                .vm_pk
+                .fri_params
+                .max_constraint_degree(),
+            ..Default::default()
+        };
+        sdk = Sdk::new(app_config.clone())?
```

Comment:
a little weird to override `sdk` versus either setting something or asserting the config matches.
I prefer if we error or panic to catch unexpected config mismatches